File: hazelcast-guides/event-driven-microservices-lab/common/src/main/java/hazelcast/platform/labs/payments/domain/Card.java
package hazelcast.platform.labs.payments.domain;

import com.github.javafaker.Faker;

public class Card {
    String cardNumber;

    public String getCardNumber() {
        return cardNumber;
    }

    public void setCardNumber(String cardNumber) {
        this.cardNumber = cardNumber;
    }

    @Override
    public String toString() {
        return "Card{" +
                "cardNumber='" + cardNumber + '\'' +
                '}';
    }

    private static Faker faker = new Faker();
    public static Card fake(){
        Card result = new Card();
        String cc = faker.finance().creditCard();
        while(cc.length() != 19)
            cc = faker.finance().creditCard();

        result.setCardNumber(cc);
        return result;
    }

}

File: hazelcast-guides/event-driven-microservices-lab/event-generator/card.py
import typing

from hazelcast.serialization.api import CompactSerializer, CompactReader, CompactWriter, CompactSerializableType


class Card:
    def __init__(self, card_number: str):
        self.card_number = card_number


class CardSerializer(CompactSerializer[Card]):
    def read(self, reader: CompactReader) -> Card:
        card_number = reader.read_string("cardNumber")
        return Card(card_number)

    def write(self, writer: CompactWriter, card: Card) -> None:
        writer.write_string("cardNumber", card.card_number)

    def get_type_name(self) -> str:
        return "hazelcast.platform.labs.payments.domain.Card"

    def get_class(self) -> typing.Type[CompactSerializableType]:
        return Card
